tests/integration-tests/tests/cloudwatch_logging/test_cloudwatch_logging.py

```diff
@@ -220,6 +220,9 @@ def _add_login_instance(self, instance):
     def _get_initial_cluster_log_state(self):
         """Get EC2 instances belonging to this cluster. Figure out their roles in the cluster."""
         for instance in cw_logs_utils.get_ec2_instances():
+            # Ensure the instance is in 'running' state
```

Comment:
If we loop using this condition, we can miss those instances which are not running but are part of the cluster.

For ex. a login node instance is not yet in running state, but its part of this cluster we create. Then we never track that instance as it will just go to the next iteration of the loop.

Reply:
The Cluster would not be in CREATE_COMPLETE state if the HeadNode is not running.
Are we not checking if the cluster creation is completed before we log in. Also The HN is the one which is the 1st instance to come up during a cluster creation so we can log into the HN even before the cluster is fully created.

Reply:
Also this change will also affect test_cloudwatch_logging's tests

Reply:
Ohh I think I misinterpreted your question.

Reply:
Do we know why these instances are not in running state? I guess the solution makes sense that we do not want to log instances that are not in running state. Can you check if it affects the other test i mentioned

Reply:
Plus I am not sure why we event run that command, as the hostname is the same as PrivateDnsName. If we dont get the PrivateDnsName whats the point of running that command.

Reply:
> The Cluster would not be in CREATE_COMPLETE state if the HeadNode is not running.

In some cases, LoginNode may not be in running state. I explained in description.


> Do we know why these instances are not in running state?

Yes, I explained in description.

> Plus I am not sure why we event run that command, as the hostname is the same as PrivateDnsName. If we dont get the PrivateDnsName whats the point of running that command.

I am not sure either. I will check and reply later. Thank you!

> Can you check if it affects the other test i mentioned?

I checked, only affects test_dcv and test_cloudwatch_logging.

Reply:
> Plus I am not sure why we event run that command, as the hostname is the same as PrivateDnsName. 

Seems there are some cases that the FQDN obtained by running `hostname -f` on the instance may differ from the `PrivateDnsName` provided by AWS EC2 instance metadata. I am not sure how to test it. Are you sure they will be totally the same in parallelcluster?

But anyway, if these two tests are stable previously, that means no instance not in `running` state was included. So it will safe to add these lines. Or at least, I can add them under `if tags.get("Name", "") == "LoginNode":`. What do you think?

Reply:
Ltes add the condition of LoginNode

Reply:
Done!